libraries/anvill_passes/src/BranchAnalysis.cpp

```diff
@@ -0,0 +1,432 @@
+    }
+
+    return std::nullopt;
+  }
+
+  z3::expr create_flag_for_name(z3::context &cont, std::string name) {
+    return cont.bool_const(name.c_str());
+  }
+
+  std::pair<z3::expr, z3::expr> symbols_as_z3(z3::context &cont,
+                                              Environment &env) {
+    auto v1name = EnvironmentBuilder::getValueName(0);
+    auto v2name = EnvironmentBuilder::getValueName(1);
+    auto v1 = cont.bv_const(
+        v1name.c_str(), this->symbols->first->getType()->getIntegerBitWidth());
+    auto v2 = cont.bv_const(
+        v2name.c_str(), this->symbols->first->getType()->getIntegerBitWidth());
+
+    env.insert(v1name, v1);
+    env.insert(v2name, v2);
+
+    return std::make_pair(v1, v2);
+  }
+
+  z3::expr get_expr_for_binop(llvm::Value *v, const Environment &env) {
+    if (v == this->symbols->first) {
+      return env.lookup(EnvironmentBuilder::getValueName(0));
+    } else if (v == this->symbols->second) {
+      return env.lookup(EnvironmentBuilder::getValueName(1));
+    }
+
+    throw std::invalid_argument(
+        "Expression Builder has value in flagdef that is not one of the symbols");
+  }
+
+
+  static z3::expr get_constant_in_z3(z3::context &cont, llvm::Type *ty,
+                                     uint64_t constant) {
+    auto bits = AtomIntExpr::GetBigEndianBits(
+        llvm::APInt(ty->getIntegerBitWidth(), constant));
+    return cont.bv_val(ty->getIntegerBitWidth(), bits.get());
+  }
+
+
+  static std::optional<z3::expr>
+  get_overflow_flag(z3::context &cont, z3::expr binop_res, z3::expr lhs,
+                    z3::expr rhs, FlagDefinition flagdef) {
+
+
+    z3::expr zero =
+        EnvironmentBuilder::get_constant_in_z3(cont, flagdef.resultTy, 0);
+    z3::expr sign_lhs = z3::slt(lhs, EnvironmentBuilder::get_constant_in_z3(
+                                         cont, flagdef.lhs->getType(), 0));
+    z3::expr sign_rhs = z3::slt(rhs, EnvironmentBuilder::get_constant_in_z3(
+                                         cont, flagdef.rhs->getType(), 0));
+    z3::expr binop_sign = z3::slt(
+        binop_res,
+        EnvironmentBuilder::get_constant_in_z3(cont, flagdef.resultTy, 0));
+    // TODO(ian): need overflow semantics for mul
+    switch (flagdef.binop) {
+      case Z3Binop::ADD:
+        return (sign_lhs ^ binop_sign) && (sign_rhs ^ binop_sign);
+      case Z3Binop::SUB:
+        return (sign_lhs ^ sign_rhs) && (sign_lhs ^ binop_sign);
+
+      default: return std::nullopt;
+    }
+  }
+
+  static std::optional<z3::expr>
+  get_carry_flag(z3::context &cont, z3::expr binop_res, z3::expr lhs,
+                 z3::expr rhs, FlagDefinition flagdef) {
+
+    switch (flagdef.binop) {
+      case Z3Binop::ADD:
+        return z3::ult(binop_res, lhs) || z3::ult(binop_res, rhs);
+      case Z3Binop::SUB: return z3::ult(lhs, rhs);
+
+      default: return std::nullopt;
+    }
+  }
+
+
+  std::optional<z3::expr>
+  get_flag_assertion(z3::context &cont, z3::expr binop_res, z3::expr lhs,
```

Comment:
I honestly can't figure out your naming scheme for functions/methods. Sometimes you use `CapitalCase`, other times `camelCase`, other times `underscore_case`. Most of anvill/remill use `CapitalCase`. Choose one and stick to it.

Reply:
The intended casing scheme is per the DrLojekyll style guide "Almost all symbols that would traditionally be visible across function or module boundaries follow CapitalCase" where symbols that do not cross module boundaries are snake cased. I can go ahead and convert all methods to CapitalCase

Reply:
Huh did I write that? I think I meant that to be stricter than your interpretation of it... oh well. Let's just merge this.